test/gpu_test.py

```diff
@@ -142,30 +142,18 @@ def test_a100_20gb_gpu_unsupported():
         stub.function(gpu=modal.gpu.A100(memory=20))(dummy)
 
 
-A100_GPU_COUNT_MAPPING = {1: api_pb2.GPU_TYPE_A100, **{i: api_pb2.GPU_TYPE_A100 for i in range(2, 5)}}
-
-
-@pytest.mark.parametrize("count,gpu_type", A100_GPU_COUNT_MAPPING.items())
-def test_gpu_type_selection_from_count(client, servicer, count, gpu_type):
+@pytest.mark.parametrize("count", [1, 2, 3, 4])
+def test_gpu_type_selection_from_count(client, servicer, count):
     import modal
 
     stub = Stub()
 
-    # Functions that use A100 20GB can only request one GPU
-    # at a time.
-    with pytest.raises(ValueError):
-        with pytest.warns(DeprecationError):
-            stub.function(gpu=modal.gpu.A100(count=2, memory=20))(dummy)
-        with stub.run(client=client):
-            pass
-
-    # Task type changes whenever user asks more than 1 GPU on
-    # an A100.
```

Comment:
These don't run properly on M1?

Reply:
The tests are kind of broken, `A100()` constructor already raises a ValueError. I think in Pytest v8, the behavior of the warns() context manager changed. Anyway the test is not relevant anymore.

I had to change back to pytest v7 manually in a later commit anyway because some pytest-markdown-docs uses some internal API

Reply:
(pytest v8 was released on Jan 27, which is why this PR suddenly upgraded versions, but only for non-cached dependencies)